src/utilities.ts

```diff
@@ -44,9 +45,36 @@ export function getCurrentYear(): number {
   return new Date().getFullYear();
 }
 
-// the number of year options to include in dropdowns before and after the current year
-// ex. if the current year is 2022, and yearRange is 6, then we want to display years from 2016-2028
-export const yearRange = 6;
+// the number of year options to include in entrance year dropdown before and after the current year
+// ex. if the current year is 2022, and yearRange is 6, then the entrance year options are 2016-2028
+export const entranceYearRange = 6;
+
+// the number of year options to include in graduation year dropdown after the entrance year
+// ex. if the entrance year is 2022, and yearRange is 6, then the entrance year options are 2022-2034
+export const gradYearRange = entranceYearRange * 2;
+
+export function computeGradYears(entranceYear: string): Readonly<Record<string, string>> {
+  const semsDict: Record<string, string> = {};
+  let entranceYearNum = parseInt(entranceYear, 10);
+  if (!entranceYear) {
+    entranceYearNum = getCurrentYear();
+  }
+  for (let i = 0; i <= gradYearRange; i += 1) {
+    const yr = String(entranceYearNum + i);
+    semsDict[yr] = yr;
+  }
+  return semsDict;
+}
+
+export function computeEntranceYears(): Readonly<Record<string, string>> {
+  const semsDict: Record<string, string> = {};
+  const curYear = getCurrentYear();
+  for (let i = -entranceYearRange; i <= entranceYearRange; i += 1) {
```

Comment:
Yes, these two functions are primarily used to populate the dropdown with year options, so I need a list of all the options. I put these functions in utilities.ts so I could take advantage of them when checking that the semesters are valid. Sure, I could've just checked that the chosen year is in the dropdown is legal, but I think its safer and easier to do it this way because there's less of a risk that a future change could break the validation by altering the how we fill the dropdown.